core-primitives/node-api/api-client-extensions/src/account.rs

```diff
@@ -18,24 +18,33 @@
 use crate::ApiResult;
 use itp_types::AccountId;
 use sp_core::crypto::Pair;
+use sp_rpc::number::NumberOrHex;
 use sp_runtime::MultiSignature;
-use substrate_api_client::{Api, ExtrinsicParams, RpcClient};
+use substrate_api_client::{Api, BalancesConfig, ExtrinsicParams, FromHexString, RpcClient};
+
+use codec::Decode;
+use core::str::FromStr;
 
 /// ApiClient extension that contains some convenience methods around accounts.
 pub trait AccountApi {
 	fn get_nonce_of(&self, who: &AccountId) -> ApiResult<u32>;
-	fn get_free_balance(&self, who: &AccountId) -> ApiResult<u128>;
+	// fn get_free_balance(&self, who: &AccountId) -> ApiResult<u128>;
 }
 
-impl<P: Pair, Client: RpcClient, Params: ExtrinsicParams> AccountApi for Api<P, Client, Params>
+impl<P: Pair, Client: RpcClient, Params, Runtime> AccountApi for Api<P, Client, Params, Runtime>
 where
 	MultiSignature: From<P::Signature>,
+	Params: ExtrinsicParams<Runtime::Index, Runtime::Hash>,
+	Runtime: BalancesConfig,
+	Runtime::Hash: FromHexString,
+	Runtime::Index: Into<u32> + Decode,
+	Runtime::Balance: TryFrom<NumberOrHex> + FromStr + Into<u128>,
 {
 	fn get_nonce_of(&self, who: &AccountId) -> ApiResult<u32> {
-		Ok(self.get_account_info(who)?.map_or_else(|| 0, |info| info.nonce))
+		Ok(self.get_account_info(who)?.map_or_else(|| 0, |info| info.nonce.into()))
 	}
 
-	fn get_free_balance(&self, who: &AccountId) -> ApiResult<u128> {
-		Ok(self.get_account_data(who)?.map_or_else(|| 0, |data| data.free))
-	}
+	// fn get_free_balance(&self, who: &AccountId) -> ApiResult<u128> {
+	// 	Ok(self.get_account_info(who)?.map_or_else(|| 0, |info| info.data.free.into()))
+	// }
```

Comment:
What was your intuition behind removing this?

Reply:
That was on my account. The problem is the generic approach: While the type `AccountData` is specified in the FrameSystemConfig, it is not known how this struct will look like. So the `.free` does not work.

So either remove this function and/or introduce a less generic function FYI: the next update will introduce a trait: https://github.com/scs/substrate-api-client/blob/7e5962bd2c9149f990b34aa25ad7bc7f6f92c868/src/api/rpc_api/frame_system.rs#L31-L41

Reply:
OK, I see thanks for the clarification, I would assume that we don't need to stay as generic, but this can be discussed later: https://github.com/integritee-network/worker/issues/1170.

@BillyWooo can you please add a comment referencing the above issue?